app/models/like.rb

```diff
@@ -0,0 +1,10 @@
+class Like < ActiveRecord::Base
+  belongs_to :author, class_name: 'User'
+  belongs_to :post
+
+  def self.update_like_counter(id)
+    post = Post.find_by(id:)
+    post.likescounter = post.likes.count
+    post.save
+  end
```

Comment:
- [ ] I think it would be better if you update the above method in order it should update the likes counter for a `Post` after a `Like` has saved. you can do it by also using the ``increment``. 

- [ **OPTIONAL** ] I also suggest you use the use one of the [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) - `after_save` to call the `#update_like_counter` method every time a new post has been saved to database, then make the `update_like_counter` method `private`
```ruby
  ...

 after_save :update_like_counter

  private

  def update_like_counter
    post.increment!(:likes_counter)
  end
```